src/main/java/org/folio/dao/budget/BudgetExpenseClassDAOImpl.java

```diff
@@ -17,7 +17,7 @@ public class BudgetExpenseClassDAOImpl implements BudgetExpenseClassDAO{
 
   private static final String SELECT_BUDGET_EXPENSE_CLASSES_BY_BUDGET_ID = "SELECT budget_expense_class.jsonb "
     + "FROM %s AS budget_expense_class "
-    + "WHERE budget_expense_class.budgetid = $1";
+    + "WHERE budget_expense_class.jsonb->>'budgetId' = $1";
```

Comment:
From performance prospective we are good even for query without index because this query made for temporary table where we are not expecting big amount of data